File: Python_Codes/11. Inner Classes/AnonymousInnerClass.py
class AnonymousInnerClassExample:
    # Interface-like Task definition using a base class
    class Task:
        def perform(self):
            raise NotImplementedError("Subclasses should implement this!")

    def execute_task(self):
        # Anonymous-like inner class using inline class definition
        class TaskImpl(self.Task):
            def perform(self):
                print("Performing a simple task...")

        task = TaskImpl()
        task.perform()

# Main execution
if __name__ == "__main__":
    example = AnonymousInnerClassExample()
    example.execute_task()

File: Projects/JavaProject/src/com/Inner_Classes/AnonymousInnerClass.java
package com.Inner_Classes;

class AnonymousInnerClassExample {

    // Interface that defines a task
    interface Task {
        void perform(); // Task that needs to be performed
    }

    // Method to demonstrate the use of an anonymous inner class
    public void executeTask() {
        // ✅ Anonymous Inner Class implementation of Task
        Task task = new Task() {
            // Implementing the perform method of the Task interface
            @Override
            public void perform() {
                System.out.println("Performing a simple task...");
            }
        };

        // Execute the task
        task.perform();
    }
}

public class AnonymousInnerClass {
    public static void main(String[] args) {
        // Create an instance of the class to invoke the executeTask method
        AnonymousInnerClassExample example = new AnonymousInnerClassExample();
        example.executeTask();
    }
}
